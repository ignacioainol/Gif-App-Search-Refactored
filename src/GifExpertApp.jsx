import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Futurama']);

  const onAddCategory = (newCategory) => {

    if (categories.find(cat => cat.toLowerCase() === newCategory.toLowerCase())) return;

    setCategories([newCategory, ...categories]);
  }

  return (
    <div className="container pt-5">
      <h1>Gif Search</h1>
      <figcaption class="blockquote-footer animate__animated animate__backInDown">
        Developed by <cite title="ignacio_Ainol">ignacio_Ainol</cite>
      </figcaption>

      <AddCategory
        onNewCategory={(value) => onAddCategory(value)}
      />

      {
        categories.map((category) =>
        (
          <GifGrid
            category={category}
            key={category} />
        ))
      }


    </div>
  )
}
