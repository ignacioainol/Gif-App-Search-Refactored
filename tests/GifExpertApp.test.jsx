import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Test en <GifExpertApp', () => {

  test('Debe hacer match con el snapshot', () => {

    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test('Cambiar el valor del input', () => {
    render(<GifExpertApp />)
    screen.debug();
    const input = screen.getByRole('textbox');

    fireEvent.input(input, { target: { value: 'Saitama' } })
    expect(input.value).toBe('Saitama');
  })

  test('Cambiar el valor del input', () => {
    render(<GifExpertApp />)
    screen.debug();
    const input = screen.getByRole('textbox');

    fireEvent.input(input, { target: { value: 'Saitama' } })
    expect(input.value).toBe('Saitama');
  });
})