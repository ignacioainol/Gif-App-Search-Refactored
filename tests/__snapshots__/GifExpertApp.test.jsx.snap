// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifExpertApp Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="container pt-5"
  >
    <h1>
      Gif Search
    </h1>
    <figcaption
      class="blockquote-footer animate__animated animate__backInDown"
    >
      Developed by 
      <cite
        title="ignacio_Ainol"
      >
        ignacio_Ainol
      </cite>
    </figcaption>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar Gifs"
        type="text"
        value=""
      />
    </form>
    <h3>
      Futurama
    </h3>
    <div
      class="spinner-border"
      role="status"
    >
      <span
        class="visually-hidden"
      >
        Loading...
      </span>
    </div>
    <div
      class="row"
    />
  </div>
</div>
`;
